src/Npgsql/PostgresDatabaseInfo.cs

```diff
@@ -201,161 +203,135 @@ internal async Task<List<PostgresType>> LoadBackendTypes(NpgsqlConnector conn, N
             if (timeout.IsSet)
                 commandTimeout = (int)timeout.CheckAndGetTimeLeft().TotalSeconds;
 
-            using var batch = new NpgsqlBatch(conn)
-            {
-                BatchCommands =
-                {
-                    new("SELECT version()"),
-                    new(GenerateLoadTypesQuery(SupportsRangeTypes, conn.Settings.LoadTableComposites)),
-                    new(GenerateLoadCompositeTypesQuery(conn.Settings.LoadTableComposites)),
-                }
-            };
+            var batchQuery = new StringBuilder()
+                .AppendLine("SELECT version();")
+                .AppendLine(GenerateLoadTypesQuery(SupportsRangeTypes, conn.Settings.LoadTableComposites))
+                .AppendLine(GenerateLoadCompositeTypesQuery(conn.Settings.LoadTableComposites));
 
             if (SupportsEnumTypes)
-                batch.BatchCommands.Add(new(GenerateLoadEnumFieldsQuery(HasEnumSortOrder)));
-
-            batch.Timeout = commandTimeout;
-            batch.AllResultTypesAreUnknown = true;
+                batchQuery.AppendLine(GenerateLoadEnumFieldsQuery(HasEnumSortOrder));
 
+            conn.WriteBuffer.Timeout = TimeSpan.FromSeconds(commandTimeout);
```

Comment:
What's the point of setting a timeout here since it's going to be replaced in `timeout.CheckAndApply` anyway?

Reply:
Yeah, you're right.
I misread the code in `timeout.CheckAndApply()`.
This is a pointless double setting.